linera-core/src/client.rs

```diff
@@ -199,7 +207,11 @@ impl MessagePolicy {
 /// * The chain being operated is called the "local chain" or just the "chain".
 /// * As a rule, operations are considered successful (and communication may stop) when
 /// they succeeded in gathering a quorum of responses.
-pub struct ChainClient<ValidatorNodeProvider, Storage> {
+pub struct ChainClient<ValidatorNodeProvider, Storage>
+where
+    Storage: linera_storage::Storage,
+    ViewError: From<Storage::ContextError>,
+{
```

Comment:
Those might not be needed on the struct definitions, only on the `impl` blocks?

Reply:
Unfortunately it is, because of the `where` clause in `ChainStateView` (see discussion in previous comment).